src/System.Xml.XmlSerializer/src/System/Xml/Serialization/NameTable.cs

```diff
@@ -78,7 +79,11 @@ internal ICollection Values
         internal Array ToArray(Type type)
         {
             Array a = Array.CreateInstance(type, _table.Count);
-            _table.Values.CopyTo(a, 0);
+            int arrayIndex = 0;
+            foreach (var value in _table.Values)
+            {
+                a.SetValue(value, arrayIndex++);
+            }
             return a;
```

Comment:
@khdang, I believe it would be possible to cast _table.Values it to ICollection to use the explicit interface implementation here.